mongo/job_store.go

```diff
@@ -133,6 +133,40 @@ func (m *JobStore) CreateTask(ctx context.Context, jobID string, taskName string
 	return newTask, err
 }
 
+// GetTask retrieves the details of a particular task, from the collection, specified by its task name and associated job id
+func (m *JobStore) GetTask(ctx context.Context, jobID string, taskName string) (models.Task, error) {
+	s := m.Session.Copy()
+	defer s.Close()
+	log.Info(ctx, "getting task from mongo DB", log.Data{"jobID": jobID, "taskName": taskName})
+
+	// If an empty jobID or taskName was passed in, return an error with a message.
+	if jobID == "" {
+		return models.Task{}, ErrEmptyIDProvided
+	} else if taskName == "" {
+		return models.Task{}, ErrEmptyTaskNameProvided
+	}
+
+	var job models.Job
+	job, err := m.findJob(s, jobID, job)
```

Comment:
You can collapse line 149 into 150 and we are not using job variable so we can throw it away (uses less memory):
```suggestion
	_, err := m.findJob(s, jobID, models.Job{})
```